### PR TITLE
#166 manage inventory

### LINKED ISSUE #166: As an owner, I want to manage inventory after creating event to be convenient
Add, edit and delete items from inventory

### PR DESCRIPTION
**Pull request by Andriets**
Close #166

### CODE REVIEW

File: EventsExpress.Core/Services/InventoryService.cs [hunk 1]
```diff
@@ -44,6 +45,26 @@ public async Task<OperationResult> AddInventar(Guid eventId, InventoryDTO invent
             }
         }
 
+        public async Task<OperationResult> DeleteInventar(Guid id)
+        {
+            var i = _context.Inventories.Find(id);
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/Services/InventoryService.cs]: Please, name variables using descriptive names.

File: EventsExpress.Core/Services/InventoryService.cs [hunk 2]
```diff
@@ -34,6 +34,7 @@ public async Task<OperationResult> AddInventar(Guid eventId, InventoryDTO invent
             {
                 var entity = _mapper.Map<InventoryDTO, Inventory>(inventoryDTO);
                 entity.EventId = eventId;
+                entity.UnitOfMeasuring = null;
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/Services/InventoryService.cs]: Why do we have to explicitly set it to null? Is it not null when coming from mapper?

File: EventsExpress.Db/EF/AppDbContext.cs [hunk 3]
```diff
@@ -178,6 +178,20 @@ protected override void OnModelCreating(ModelBuilder builder)
...
+            builder.Entity<UnitOfMeasuring>()
+                .HasData(
+                new UnitOfMeasuring[]
+                {
+                    new UnitOfMeasuring { Id = Guid.NewGuid(), UnitName = "Units", ShortName = "u"},
+                    new UnitOfMeasuring { Id = Guid.NewGuid(), UnitName = "Kilograms", ShortName = "kg"},
+                    new UnitOfMeasuring { Id = Guid.NewGuid(), UnitName = "Grams", ShortName = "g"},
+                    new UnitOfMeasuring { Id = Guid.NewGuid(), UnitName = "Liters", ShortName = "l"},
+                    new UnitOfMeasuring { Id = Guid.NewGuid(), UnitName = "Miliiters", ShortName = "ml"},
+                    new UnitOfMeasuring { Id = Guid.NewGuid(), UnitName = "Metre", ShortName = "m"},
```
Comment (iuriikhrystiuk) [inline EventsExpress.Db/EF/AppDbContext.cs]: Please use plural as above `Meters` and `Centimeters`

File: EventsExpress/ClientApp/src/components/inventory/InventoryList.js [hunk 4]
```diff
@@ -1,61 +1,190 @@
...
+            disabledEdit: false
         };
 
         this.handleOnClickCaret = this.handleOnClickCaret.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+        this.onCancel = this.onCancel.bind(this);
+    }
+    
+    componentDidMount() {
+        console.log('mount');
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/inventory/InventoryList.js]: Remove the console.log.

File: EventsExpress/ClientApp/src/components/inventory/InventoryList.js [hunk 5]
```diff
@@ -1,61 +1,190 @@
...
-                                                    
-                                </tbody>
-                            </table>
+                <div>
+                    {event.user.id === user.id 
+                    ?  <IconButton
+                            disabled = {this.state.disabledEdit}
+                            onClick = {this.addItemToList.bind(this)}
+                            size = "small">
+                            <span class="icon"><i class="fa-sm fas fa-plus"></i></span> &nbsp; Add item &nbsp;
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/inventory/InventoryList.js]: Do you really need these & characters here?

File: EventsExpress/ClientApp/src/containers/event-item-view.js [hunk 6]
```diff
@@ -33,28 +35,31 @@ class EventItemViewWrapper extends Component{
...
+                inventories={this.props.inventories} 
                 event={this.props.event}
                 match={this.props.match} 
                 onLeave={this.onLeave} 
                 onJoin={this.onJoin}
                 onCancel={this.onCancel}
                 onApprove={this.onApprove}
-                current_user={this.props.current_user} 
+                current_user={this.props.current_user}
+                get_inventories_by_event_id={this.props.get_inventories_by_event_id}
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/containers/event-item-view.js]: Why do you need to pass this method from here if your component is already connected to state? Consider moving it there.

File: EventsExpress/ClientApp/src/containers/event-item-view.js [hunk 7]
```diff
@@ -33,28 +35,31 @@ class EventItemViewWrapper extends Component{
...
+                get_inventories_by_event_id={this.props.get_inventories_by_event_id} 
             />
     }
 }
 
 const mapStateToProps = (state) => ({
     event: state.event, 
-    current_user: state.user
+    current_user: state.user,
+    inventories: state.inventories
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/containers/event-item-view.js]: Please consider not passing inventories from here, but connecting it directly in the component.

File: EventsExpress/ClientApp/src/services/EventsExpressService.js [hunk 8]
```diff
@@ -288,6 +288,49 @@ export default class EventsExpressService {
         return res;
     }
 
+    setEvent = async (data) => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/services/EventsExpressService.js]: There was one method AFAIR for saving events. Why a new one?

File: EventsExpress.Core/Services/InventoryService.cs (hunk 2 above)
Comment (Andriets) [reply EventsExpress.Core/Services/InventoryService.cs]: Now it null when coming from mapper

File: EventsExpress/ClientApp/src/components/inventory/InventoryList.js (hunk 5 above)
Comment (Andriets) [reply EventsExpress/ClientApp/src/components/inventory/InventoryList.js]: Yes, it's needed for a space

File: EventsExpress/ClientApp/src/services/EventsExpressService.js (hunk 8 above)
Comment (Andriets) [reply EventsExpress/ClientApp/src/services/EventsExpressService.js]: There is only one method for saving events, i didn't do more

File: EventsExpress/ClientApp/src/components/inventory/InventoryList.js [hunk 9]
```diff
@@ -1,61 +1,190 @@
...
+        }];
+        
+        this.props.get_inventories(updateList);
+        this.setState({
+            disabledEdit: true
+        });
+    }
+
+    deleteItemFromList = inventar => {
+        console.log('before deleting', this.props.eventId);
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/inventory/InventoryList.js]: Remove **ALL** console log statements.

File: EventsExpress/ClientApp/src/containers/event-item-view.js [hunk 10]
```diff
@@ -3,12 +3,14 @@ import { connect } from 'react-redux';
...
 import Spinner from '../components/spinner';
 import get_event from '../actions/event-item-view';
+import { get_inventories_by_event_id }  from '../actions/inventory-list';
 import { join, leave, resetEvent, cancel_event, approveUser } from '../actions/event-item-view';
 
 class EventItemViewWrapper extends Component{
     componentWillMount(){    
         const { id } = this.props.match.params;
         this.props.get_event(id);
+        this.props.get_inventories_by_event_id(id);
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/containers/event-item-view.js]: Why do you still load these invetories here? I thought you are doing that in the component already?

File: EventsExpress/ClientApp/src/services/EventsExpressService.js [hunk 11]
```diff
@@ -288,6 +288,49 @@ export default class EventsExpressService {
         return res;
     }
 
+    setEventTemplate = async (data) => {
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/services/EventsExpressService.js:246]: I thought `setEventTemplate` already exists. Why are you adding a new one?

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**